.github/workflows/ci.yml

```diff
@@ -55,8 +55,8 @@ jobs:
       - name: Install cargo-lambda
         run: pip install cargo-lambda
 
-      - name: Install protobuf-compiler
-        run: sudo apt-get install -y protobuf-compiler
+      - name: Install Protoc
```

Comment:
Hi,
Are these changes enough for these requirements?
"
protobuf-compiler needs to be installed for all build steps and platforms. Currently main is broken because protobuf compiler isn't installed on all platforms.

Expectations

Builds should pass on all platforms.
Use https://github.com/marketplace/actions/setup-protoc instead of manually installing
Reduce duplication as much as possible in ci.yml
"
@coderabbitai